sqlx-core/src/sqlite/connection/mod.rs

```diff
@@ -62,9 +62,12 @@ impl Connection for SqliteConnection {
 
     type Options = SqliteConnectOptions;
 
-    fn close(self) -> BoxFuture<'static, Result<(), Error>> {
-        // nothing explicit to do; connection will close in drop
-        Box::pin(future::ok(()))
+    fn close(mut self) -> BoxFuture<'static, Result<(), Error>> {
+        Box::pin(async move {
+            self.statements.clear();
+            self.statement.take();
+            self.worker.shutdown().await
```

Comment:
If you make `StatementWorker::shutdown()` work like `StatementWorker::reset()` does and have it return `impl Future` instead of `.await`ing it directly, then you could do something like:

```rust
let fut = self.worker.shutdown();

// drop the database handle
drop(self);

// wait for the statement worker to exit
fut.await
```

Reply:
Can do.. any idea what's up with the actix memory access error?